core/src/main/scala/org/apache/spark/sql/delta/files/CdcAddFileIndex.scala

```diff
@@ -39,7 +39,8 @@ class CdcAddFileIndex(
     filesByVersion: Seq[CDCDataSpec[AddFile]],
     deltaLog: DeltaLog,
     path: Path,
-    snapshot: Snapshot
+    snapshot: Snapshot,
+    override val rowIndexFilters: Option[Map[String, RowIndexFilterType]] = None
```

Comment:
docs for the new param. Also mention the details such as the path should be a URI encoded in order for the look up to work.

Reply:
same for remove. Also do we want to add assert to check if there is one entry for each file?

Reply:
Done adding docs.

Reply:
It's not necessary that `rowIndexFilters` contains all files in `filesByVersion`, the former is a superset of the latter. IMO asserting here won't bring too much add value.